packages/flutter_tools/test/general.shard/cache_test.dart

```diff
@@ -562,6 +562,36 @@ void main() {
   }, overrides: <Type, Generator> {
     Platform: () => FakePlatform()..operatingSystem = 'fuchsia',
   });
+
+  testUsingContext('macOS desktop artifacts ignore filtering when requested', () {
+    final MockCache mockCache = MockCache();
+    final MacOSEngineArtifacts artifacts = MacOSEngineArtifacts(mockCache);
+    when(mockCache.includeAllPlatforms).thenReturn(false);
+    when(mockCache.platformOverrideArtifacts).thenReturn(<String>{'macos'});
+    expect(artifacts.getBinaryDirs(), isNotEmpty);
+  }, overrides: <Type, Generator> {
+    Platform: () => FakePlatform()..operatingSystem = 'linux',
```

Comment:
nit: prefer constructor instead of setter

```suggestion
    Platform: () => FakePlatform(operatingSystem: 'linux'),
```